R/animint.R

```diff
@@ -962,26 +938,47 @@ animint2dir <- function(plot.list, out.dir = tempfile(),
   }
 
   ## Extract essential info from ggplots, reality checks.
-  ggplot.list <- list()
-  AllPlotsInfo <- list()
   for(list.name in names(plot.list)){
     p <- plot.list[[list.name]]
     if(is.ggplot(p)){
+      ## Save original mapping
+      for(layer_i in seq_along(p$layers)){
+        ## Viz has not been used before
+        if(is.null(p$layers[[layer_i]]$orig_mapping)){
+          p$layers[[layer_i]]$orig_mapping <- 
+            if(is.null(p$layers[[layer_i]]$mapping)){
+              p$mapping
+            }else{
+              p$layers[[layer_i]]$mapping
+            }
+        }else{
+          p$layers[[layer_i]]$mapping <- p$layers[[layer_i]]$orig_mapping
+        }
+      }
+      
       ## Before calling ggplot_build, we do some error checking for
       ## some animint extensions.
       checkPlotForAnimintExtensions(p, list.name)
-      
-      ## If plot is correct, save to meta for further processing
-      parsed_info <- parsePlot(meta, p, list.name) # calls ggplot_build.
-      AllPlotsInfo[[list.name]] <- parsed_info[[1]]
-      ggplot.list[[list.name]] <- parsed_info[2:3]
     }else if(is.list(p)){ ## for options.
       meta[[list.name]] <- p
     }else{
       stop("list items must be ggplots or option lists, problem: ", list.name)
     }
   }
-
+  
+  ## Call ggplot_build in parsPlot for all ggplots
+  ggplot.list <- list()
+  AllPlotsInfo <- list()
+  for(list.name in names(plot.list)){
+    p <- plot.list[[list.name]]
+    if(is.ggplot(p)){
+      ## If plot is correct, save to meta for further processing
+      parsed_info <- parsePlot(meta, p, list.name) # calls ggplot_build.
+      AllPlotsInfo[[list.name]] <- parsed_info[[1]]
+      ggplot.list[[list.name]] <- parsed_info[2:3]
```

Comment:
it would be easier to read/understand if you use `$name` rather than `[[1]]` or `[2:3]`